app/service/agent_svc.py

```diff
@@ -65,8 +75,21 @@ def __init__(self):
         :param status:
         :return: a JSON status message
         """
-        await self.get_service('data_svc').create('core_result', dict(link_id=link_id, output=output))
-        await self.get_service('data_svc').update('core_chain', key='id', value=link_id,
+        link = await self.get_service('data_svc').explode_chain(criteria=dict(id=link_id))
+        op = (await self.get_service('data_svc').dao.get('core_operation', dict(id=link[0]['op_id'])))[0]
```

Comment:
let's drop the change to this function from this PR. I can see where you're going with it-- which is interesting, so I want to separate this from the rest of the PR... reasons:

I want to think through whether we should/shouldn't accept posted results from untrusted agents. I'm not sure yet and want to get the core untrusted logic in place so we can test drive it before deciding.

Reply:
I understand, I left the last seen more accurate for the moment.
This is because it could happen that an agent busy executing many links and sending results, postpone sending the heartbeat signal (if I understand correctly ...).
If we don't update the last_seen here, it could happen (even if remote case) that the sniffer function considers it untrusted, when in reality it is not.
What do you think about it?